src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private Object[] array;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void checkArraySize() {
+        if (array.length == size) {
+            Object[] addObject = array;
+            array = new Object[size * 3 / 2 + 1];
+            System.arraycopy(addObject, 0, array, 0, addObject.length);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        checkArraySize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            checkArraySize();
+            System.arraycopy(array, index, array,index + 1, size - index);
+            array[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can't add. Not appropriate index. ");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            checkArraySize();
+            array[size] = list.get(i);
+            size++;
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return (T) array[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can't get. Not appropriate index.");
+        }
     }
 
     @Override
     public void set(T value, int index) {
+        if (index >= 0 && index < size) {
+            array[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can't set. Not appropriate index.");
+        }
 
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            final T value = (T) array[index];
+            System.arraycopy(array, index + 1, array, index, size - index - 1);
+            size--;
+            array[size] = null;
+
+            return value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can't remove. Not appropriate index.");
```

Comment:
```suggestion
        } 
        throw new ArrayListIndexOutOfBoundsException("Can't remove. Not appropriate index.");
```
після return в if блоці не потрібен else